pkg/memlimit/memlimit.go

```diff
@@ -0,0 +1,54 @@
+package memlimit
+
+import (
+	"fmt"
+	"os"
+	"runtime/debug"
+	"strconv"
+
+	"github.com/stackrox/rox/pkg/logging"
+)
+
+var log = logging.LoggerForModule()
+
+func SetMemoryLimit() {
+	var limit string
+	var reason string
+
+	defer func() {
```

Comment:
Instead of this defer, let's create a func called getMemLimit() that returns the derived memory limit and then does this

Reply:
then we can test this function https://github.com/stackrox/stackrox/pull/9064#discussion_r1432678704

Reply:
The defer just prints logs. The only real work is done when `ROX_MEMLIMIT` is set which is outside the `defer`

Reply:
I kept the `defer`, but I made it testable via `var setMemoryLimit = debug.SetMemoryLimit` and then setting a new value in the test

Reply:
> The defer just prints logs. The only real work is done when `ROX_MEMLIMIT` is set which is outside the `defer`

Not sure why I didn't understand the suggestion until now. Ignore this message ^. That being said, I kept it as-is, but made it testable. Let me know if it's still preferred to refactor

Reply:
Honestly, I really do prefer the refactor rather than a very large defer